artemis-server/src/main/java/org/apache/activemq/artemis/core/server/metrics/MetricsManager.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.activemq.artemis.core.server.metrics;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.ToDoubleFunction;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.binder.jvm.JvmMemoryMetrics;
+import org.apache.activemq.artemis.api.core.management.ResourceNames;
+import org.apache.activemq.artemis.core.server.ActiveMQServerLogger;
+
+public class MetricsManager {
+
+   private String brokerName;
+
+   private MeterRegistry meterRegistry;
+
+   private Map<String, List<Meter>> meters = new HashMap<>();
+
+   public MetricsManager(String brokerName, ActiveMQMetricsPlugin metricsPlugin) {
+      this.brokerName = brokerName;
+      meterRegistry = metricsPlugin.getRegistry();
+      Metrics.globalRegistry.add(meterRegistry);
+      new JvmMemoryMetrics().bindTo(meterRegistry);
+   }
+
+   public MeterRegistry getMeterRegistry() {
+      return meterRegistry;
+   }
+
+   public void registerQueueGauge(String metricName, String address, String queue, Object object, ToDoubleFunction f, String description) {
```

Comment:
This isn't possible because a `QueuePendingMessageMetrics` isn't always passed into the method (e.g. for all the queue metrics that aren't tracked by the `QueuePendingMessageMetrics` object).